main.tf

```diff
@@ -47,13 +47,13 @@ data "aws_iam_policy_document" "ami_backup" {
 data "archive_file" "ami_backup" {
   type        = "zip"
   source_file = "${path.module}/ami_backup.py"
-  output_path = "${path.module}/ami_backup.zip"
+  output_path = ".terraform/terraform-aws-ec2-ami-backup/backup-${md5(file("${path.module}/ami_backup.py"))}.zip"
```

Comment:
I'm concerned with the portability of hardcoding both `.terraform` and the module name `terraform-aws-ec2-ami-backup` which is subject to change in forks. 

I propose the following:
- [ ] create a new variable called `tfstate_dir` and default it to `.terraform`
- [ ] use a `local` called `module_name` which is derived from `basename(path.module)`
- [ ] use a `local` to define an `output_dir` which is equal to `${tfstate_dir}/${module_name}`

Reply:
This is also a nice fix: https://github.com/hashicorp/terraform/issues/8204#issuecomment-332239294

Take your pick =)